External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -189,6 +192,45 @@ namespace FEXCore::Context {
     return NewThreadState;
   }
 
+  bool Context::HandleSegfault(FEXCore::Core::InternalThreadState *Thread, int Signal, void *info, void *ucontext) {
+    auto CTX = Thread->CTX;
+
+    auto FaultAddress = (uintptr_t)((siginfo_t*)info)->si_addr;
+
+    bool DoFlush = false;
+
+    {
+      FHU::ScopedSignalMaskWithSharedMutex lk(CTX->MemoryEntryMutex, true);
+
+      auto Entry = CTX->MemoryMaps.upper_bound(FaultAddress);
+
+      if (Entry != CTX->MemoryMaps.begin()) {
+        --Entry;
+
+        if (Entry->first <= FaultAddress && (Entry->first + Entry->second.Length) > FaultAddress) {
```

Comment:
Does the width of the memory access that triggered the segfault need to be factored into this computation?

Reply:
No, as that will trigger two segfaults if it is a cross page access. Added a comment to make this clear.